staging/src/k8s.io/client-go/tools/events/interfaces.go

```diff
@@ -45,6 +45,12 @@ type EventBroadcaster interface {
 	// NewRecorder returns an EventRecorder that can be used to send events to this EventBroadcaster
 	// with the event source set to the given event source.
 	NewRecorder(scheme *runtime.Scheme, reportingController string) EventRecorder
+
+	// StartEventWatcher enables you to watch for emitted events without usage
+	// of StartRecordingToSink. This lets you also process events in a custom way (e.g. in tests).
+	// NOTE: events received on your eventHandler should be copied before being used.
+	// TODO: figure out if this can be removed.
```

Comment:
This shouldn't be part of this interface - this case easily be a helper function in utils.go or sth like that.

Reply:
I'd leave it here for now to merge this PR. 

I can send then a PR to refactor the testing, as it relies on events being emitted.

Reply:
It's negligible refactor - unless I'm not missing something, the only thing you need is to pass a watch.Interface to this method and change the 3 calls to that function to pass e.Watch() - seems like 10 lines of code.